accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,320 @@
+- Feature Name: Ansible Integration
+- Start Date: 2021-02-01
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes how to integrate and manage your existing Ansible nodes in Uyuni with the help of Salt and its "Ansible-Gate" module. This means that you would be able to import Ansible inventories, their systems and also apply playbooks to these nodes using Uyuni.
+
+
+# Motivation
+[motivation]: #motivation
+
+There are two main motivations for this:
+
+1. A user might have some investment in Ansible in the past but wants to switch to Uyuni now.
+This RFC would offer a transition path for that. The user can import the Ansible systems, start with the already existing playbooks, get familiar with Uyuni and then switch over. It is also possible to manage clients with Salt and Ansible in parallel.
+2. Managing parts of the infrastructure in Ansible.
+If it is not possible or not wanted to move everything to Salt, it is possible to just keep using Ansible for the few clients that cannot be transitioned.
+
+This RFC is more focused on allowing users to import their existing Ansible environments into Uyuni, allow some basic operations, coexistence of Salt and Ansible managed infrastructure in Uyuni, eventually apply playbooks and also to provide an easy way to transition from Ansible to a fully-featured Salt minion managed system in Uyuni, rather than making Uyuni a top-featured UI to build your Ansible infrastructure from scratch.
+
+
+# Detailed design
+[design]: #detailed-design
+
+There are three main parts/goals here, conceptually:
+
+1) Collect data from an Ansible control node (Inventory / SSH Keys / Playbooks, etc) and import the hosts in the inventory as ANSIBLE systems in Uyuni.
+2) Operate Ansible: Apply playbooks via control node node / Apply Salt commands & states directly to Ansible managed systems / Make Ansible system become a fully-featured Salt Minion.
+3) Maintain Ansible infrastructure: Own Uyuni Playbook catalog, revisions, formulas.
+
+## Collecting data from an Ansible control node
+
+This section elaborates how Uyuni would do for gathering information for an Ansible control node. There are different scenarios that we can consider here:
+
+1. Adding an external host Ansible control node:
+
+The premise here is that there is already an existing and configured Ansible infrastructure somewhere and we want to import it into Uyuni. This Ansible control node is not even a registered system in Uyuni. The hosts defined on the inventory can be imported Uyuni and will be displayed as "Foreign/ANSIBLE" or "Salt/ANSIBLE", etc. The "ANSIBLE" entitlement means that host is being managed by an Ansible control node.
+
+Here we're **not** primarely focused on building a new Ansible managed infrastructure from scratch using Uyuni.
+
+NOTE: Currently Ansible is not shipped in SLE15 any SP. In order to allow an SLE Ansible control node, we would need to ship Ansible in probably in SLE15 client tools.
```

Comment:
IIUC in this hypothesis the Uyuni Server would display Ansible-managed systems obtained from an inventory.

What mechanism/protocol would be used to access such inventory?

Would there be any feature other than listing of Ansible-managed clients?